KokkosAlgorithms: CMake issue in unit tests

I got the following CMake error when enabling tests for KokkosAlgorithms in the Trilinos build:
```
CMake Error at cmake/tribits/core/package_arch/TribitsAddExecutable.cmake:471 (MESSAGE):
  ERROR: 'kokkos_gtest' in TESTONLYLIBS not a TESTONLY lib! If this a regular
  library in this SE package or in an dependent upstream SE package then
  TriBITS will link automatically to it.  If you remove this and it does not
  link, then you need to add a new SE package dependency to this SE package's
  dependencies file
  /scratch/prj/Trilinos/Trilinos/packages/kokkos/algorithms/cmake/Dependencies.cmake
Call Stack (most recent call first):
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:68 (TRIBITS_ADD_EXECUTABLE)
  cmake/tribits/core/package_arch/TribitsAddExecutableAndTest.cmake:213 (TRIBITS_ADD_EXECUTABLE_WRAPPER)
  packages/kokkos/algorithms/unit_tests/CMakeLists.txt:31 (TRIBITS_ADD_EXECUTABLE_AND_TEST)
```
When I deleted the TESTONLYLIBS line in `kokkos/algorithms/unit_tests/CMakeLists.txt`, the error went away.  I will patch in Trilinos for now and then submit a Kokkos PR.

Oops, deleting the `TESTONLYLIBS` line broke the build:
```
.../Trilinos/packages/kokkos/algorithms/unit_tests/UnitTestMain.cpp:44:10: fatal error: 'gtest/gtest.h' file not found
#include <gtest/gtest.h>
```

--- a/algorithms/unit_tests/CMakeLists.txt
+++ b/algorithms/unit_tests/CMakeLists.txt
@@ -3,6 +3,22 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 INCLUDE_DIRECTORIES(REQUIRED_DURING_INSTALLATION_TESTING ${CMAKE_CURRENT_SOURCE_DIR})
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../src )
 
+SET(GTEST_SOURCE_DIR ${${PARENT_PACKAGE_NAME}_SOURCE_DIR}/tpls/gtest)
+INCLUDE_DIRECTORIES(${GTEST_SOURCE_DIR})
+
+# mfh 03 Nov 2017: The gtest library used here must have a different
+# name than that of the gtest library built in KokkosCore.  We can't
+# just refer to the library in KokkosCore's tests, because it's
+# possible to build only (e.g.,) KokkosAlgorithms tests, without
+# building KokkosCore tests.
+
+TRIBITS_ADD_LIBRARY(
+  kokkosalgorithms_gtest
+  HEADERS ${GTEST_SOURCE_DIR}/gtest/gtest.h
+  SOURCES ${GTEST_SOURCE_DIR}/gtest/gtest-all.cc
+  TESTONLY
+  )
+
 SET(SOURCES
   UnitTestMain.cpp 
   TestCuda.cpp
@@ -34,5 +50,5 @@ TRIBITS_ADD_EXECUTABLE_AND_TEST(
   COMM serial mpi
   NUM_MPI_PROCS 1
   FAIL_REGULAR_EXPRESSION "  FAILED  "
-  TESTONLYLIBS kokkos_gtest
+  TESTONLYLIBS kokkosalgorithms_gtest
   )

